### PR TITLE
Display push time in status

### LINKED ISSUE #69: Means to display push times
Would be nice to easily be able to check when students submit code. Perhaps this could be done as part of `assigner status`; also maybe a flag for `assigner get`?

### PR DESCRIPTION
**Pull request by LinuxMercedes**
Also display time in a more human-readable format.

Closes #69 in part; the remainder will be left for a future feature.

### CODE REVIEW

File: assigner/commands/status.py [hunk 1]
```diff
@@ -85,16 +85,15 @@ def status(conf, args):
...
-            if commits:
-                head = commits[0]
+            if head:
                 row[6] = head["short_id"]
                 row[7] = head["author_name"]
                 created_at = head["created_at"]
                 # Fix UTC offset format in GitLab's datetime
                 created_at = created_at[:-6] + created_at[-6:].replace(':', '')
-                row[8] = datetime.strptime(created_at, "%Y-%m-%dT%H:%M:%S.%f%z")
+                row[8] = datetime.strptime(created_at, "%Y-%m-%dT%H:%M:%S.%f%z").astimezone().strftime("%c")
```
Comment (brhoades) [inline assigner/commands/status.py]: line length

File: assigner/baserepo.py [hunk 2]
```diff
@@ -253,6 +253,28 @@ def list_commits(self, ref_name="master"):
             "/projects/{}/repository/commits".format(self.id), params
         )
 
+    def list_pushes(self):
+        return self._gl_get(
+            "/projects/{}/events?action=pushed".format(self.id)
+        )
+
+    def get_last_HEAD_commit(self, ref="master"):
+        matching_pushes = list(filter(lambda push: push['push_data']['ref'] == ref, self.list_pushes()))
```
Comment (brhoades) [inline assigner/baserepo.py]: np: wrap please

File: assigner/baserepo.py [hunk 3]
```diff
@@ -253,6 +253,28 @@ def list_commits(self, ref_name="master"):
...
+        commits = self.list_commits(ref)
+
+        HEAD = None
+
+        if commits:
```
Comment (brhoades) [inline assigner/baserepo.py]: np: let's flip this to reduce the block's indentation / line length.

```python
if not commits:
    return None
```

File: assigner/baserepo.py [hunk 4]
```diff
@@ -253,6 +253,28 @@ def list_commits(self, ref_name="master"):
...
+            HEAD = commits[0]
+            # Gitlab's commit created_at time uses the git metadata;
+            # rather than trusting students, we get the time the commit was pushed at
+            if matching_pushes and HEAD['id'] == matching_pushes[0]['push_data']['commit_to']:
+                # For whatever reason, Gitlab uses a different time format here than for commits...
+                unmangled_time = matching_pushes[0]['created_at'][:-1] + "-0000"
```
Comment (brhoades) [inline assigner/baserepo.py]: Can you give an example of this? It seems like passing it through datetime may be better.

File: assigner/baserepo.py (hunk 3 above)
Comment (LinuxMercedes) [reply assigner/baserepo.py]: Can you tell I've been writing rust and dreaming of Option/Result combinators lately?

Comment (brhoades) [reply assigner/baserepo.py]: Haha maybe ;)

File: assigner/baserepo.py (hunk 4 above)
Comment (LinuxMercedes) [reply assigner/baserepo.py]: Basically the push event time looks like `"2018-01-05T20:33:21.000Z"` whereas the commit time looks like `"2018-01-05T13:44:43.000-06:00"`. I read the `strptime` docs and I don't know how to get `datetime` to parse the first date format.

File: assigner/commands/status.py (hunk 1 above)
Comment (LinuxMercedes) [reply assigner/commands/status.py]: What's the pythonic way to wrap this? I want to do something like

```
datetime
.strptime()
.astimezone()
.strftime()
```

but I'd need line continuations to make that parse right.

Comment (brhoades) [reply assigner/commands/status.py]:
```python
row[8] = datetime.strptime(
    created_at, "%Y-%m-%dT%H:%M:%S.%f%z"
).astimezone().strftime("%c") 
```
Is my personal favorite. I don't think leading with periods like that is common. I also would like that, but that's more of a ruby/js thing in my experience.

**Comment by LinuxMercedes** [comment]
@brhoades nits fixed!

**Pull request merged**